src/lib/features/playground/playground-service.ts

```diff
@@ -102,17 +102,21 @@ export class PlaygroundService {
 
         let filteredProjects: typeof projects;
         if (this.flagResolver.isEnabled('privateProjects')) {
-            const accessibleProjects =
+            const projectAccess =
                 await this.privateProjectChecker.getUserAccessibleProjects(
                     userId,
                 );
-            filteredProjects =
-                projects === ALL
-                    ? accessibleProjects
-                    : projects.filter((project) =>
-                          accessibleProjects.includes(project),
-                      );
-            console.log(accessibleProjects);
+            if (projectAccess.mode === 'all') {
+                filteredProjects = projects;
+            } else {
+                if (projects === ALL) {
```

Comment:
if you flatten those to if/else if/else I think TS will still work out the details.